mlir/lib/Dialect/Tensor/IR/TensorOps.cpp

```diff
@@ -4332,21 +4332,24 @@ LogicalResult PackOp::canonicalize(PackOp packOp, PatternRewriter &rewriter) {
           rewriter.create<tensor::CastOp>(loc, newSrcType, packOp.getSource());
     }
     Value dest = packOp.getDest();
+    Type originalResultType = dest.getType();
     if (destShape != packOp.getDestType().getShape()) {
       auto newDestType = packOp.getDestType().clone(destShape);
       dest =
           rewriter.create<tensor::CastOp>(loc, newDestType, packOp.getDest());
     }
```

Comment:
nit: we can use a boolean, then we save one comparison.

```suggestion
    bool needNewDest = destShape != packOp.getDestType().getShape();
    if (needNewDest) {
      auto newDestType = packOp.getDestType().clone(destShape);
      dest =
          rewriter.create<tensor::CastOp>(loc, newDestType, packOp.getDest());
    }
    // ...
    // Insert a cast if needed.
    if (needNewDest) {
      // ...
    }
```